Mathlib/Analysis/LocallyConvex/AbsConvex.lean

```diff
@@ -40,6 +42,99 @@ open NNReal Pointwise Topology
 
 variable {𝕜 E F G ι : Type*}
 
+section AbsolutelyConvex
+
+variable (𝕜) [SeminormedRing 𝕜] [SMul 𝕜 E] [SMul ℝ E] [AddCommMonoid E]
+/-- A set is absolutely convex if it is balanced and convex. -/
+def AbsConvex (s : Set E) : Prop := Balanced 𝕜 s ∧ Convex ℝ s
```

Comment:
Wait, do you really mean `ℝ` here and not `𝕜`?

Reply:
```
def AbsConvex (s : Set E) : Prop := Balanced 𝕜 s ∧ Convex 𝕜 s
```
with the current hypothesis would fail with `failed to synthesize OrderedSemiring 𝕜`

We could perhaps have:
```
variable (𝕜₁ 𝕜₂) [SeminormedRing 𝕜₁] [SMul 𝕜₁ E] [OrderedSemiring 𝕜₂] [SMul 𝕜₂ E] [AddCommMonoid E]
/-- A set is absolutely convex if it is balanced and convex. -/
def AbsConvex (s : Set E) : Prop := Balanced 𝕜₁ s ∧ Convex 𝕜₂ s
```
?

Reply:
What's your use case?

Reply:
@YaelDillies at this point in time I'm primarily interested in vector spaces over ℝ or ℂ, so I'd be happy to use `[RCLike 𝕜]`.

The `Analysis/LocallyConvex/AbsConvex.lean` currently in the main branch defines:
```
/-- The type of absolutely convex open sets. -/
def AbsConvexOpenSets :=
  { s : Set E // (0 : E) ∈ s ∧ IsOpen s ∧ Balanced 𝕜 s ∧ Convex ℝ s }
```
so I thought that my definition of `AbsConvex` needed to be sufficiently general that I could replace ` Balanced 𝕜 s ∧ Convex ℝ s` with it in the above.

Reply:
Does this agree with the literature?

Reply:
(Bourbaki, TVS starts with 𝕜 as a topological division ring, but switches to ℝ for Sections 2-7 of Chapter 2.)

Reply:
The definition of `AbsConvexOpenSets` seems to have been added to mathlib3 in this [commit](https://github.com/leanprover-community/mathlib3/commit/35e376288bcf74088cf17e6a764331b01162a4fc) as part of https://github.com/leanprover-community/mathlib3/pull/15035/ - I don't know what the reference was?

Reply:
@mcdoll, can you enlighten us?

Reply:
For now, @mans0954, can you clarify in the docstring that convexity and balance are taken according to different fields?

Reply:
Done.

Reply:
Sorry, I meant the docstring of the definition. It's great to also have added it to the module docs, but the docstring is what people will see first

Reply:
Done, thanks.